### PR TITLE
Pull and rebase CH vhost-user-backend commits

### PR DESCRIPTION
**Pull request by slp**
This is still needs some changes to get it up to rust-vmm's standards, but I think it's a good starting point.

### CODE REVIEW

**Comment by slp** [comment]
We'd also need some volunteers to act as code owners. I volunteer myself. @sboeuf @rbradford @alexandruag @jiangliu would you be interested?

**Comment by jiangliu** [comment]
> We'd also need some volunteers to act as code owners. I volunteer myself. @sboeuf @rbradford @alexandruag @jiangliu would you be interested?

Sure, I'm glad to help here:)

File: Cargo.toml [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
 
 [dependencies]
+epoll = ">=4.0.1"
+libc = ">=0.2.39"
+log = ">=0.4.6"
+virtio-bindings = "0.1.0"
+vm-memory = {version = ">=0.2.0", features = ["backend-mmap"]}
+vm-virtio = { git = "https://github.com/cloud-hypervisor/vm-virtio", branch = "dragonball" }
+vmm-sys-util = ">=0.3.1"
+vhost_rs = { git = "https://github.com/cloud-hypervisor/vhost", branch = "dragonball", package = "vhost", features = ["vhost-user-slave"] }
```
Comment (rbradford) [inline Cargo.toml]: Oh dear I thought this "dragonball" branch had gone, it's diverged from the "ch" branch in the same repo

File: Cargo.toml [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
 edition = "2018"
+license = "Apache-2.0"
 
 [dependencies]
+epoll = ">=4.0.1"
+libc = ">=0.2.39"
+log = ">=0.4.6"
+virtio-bindings = "0.1.0"
+vm-memory = {version = ">=0.2.0", features = ["backend-mmap"]}
+vm-virtio = { git = "https://github.com/cloud-hypervisor/vm-virtio", branch = "dragonball" }
```
Comment (rbradford) [inline Cargo.toml]: Oh dear. I missed this one as well.

**Review by rbradford — COMMENTED** [review]
Can we get the vm-virtio and vhost changes upstream first so we can depend on the rust-vmm versions instead of these forks?

**Comment by slp** [comment]
> Can we get the vm-virtio and vhost changes upstream first so we can depend on the rust-vmm versions instead of these forks?

@rbradford Either I did something wrong while creating the PR, or you're reviewing an interim commit, because with all changes applied, it should only depend on rust-vmm's crates. In fact, GitHub marks as *Outdated* your comments on `Cargo.toml`. :-?

File: Cargo.toml [hunk 3]
```diff
@@ -1,8 +1,16 @@
...
 
 [dependencies]
+epoll = ">=4.0.1"
+libc = ">=0.2.39"
+log = ">=0.4.6"
+virtio-bindings = "0.1.0"
+vm-memory = {version = ">=0.2.0", features = ["backend-mmap", "backend-atomic"]}
+vm-virtio = { git = "https://github.com/rust-vmm/vm-virtio" }
+vmm-sys-util = ">=0.3.1"
+vhost_rs = { git = "https://github.com/rust-vmm/vhost", package = "vhost", features = ["vhost-user-slave"] }
```
Comment (rbradford) [inline Cargo.toml:16]: These are almost in alphabetical order.. :-)

**Review by rbradford — APPROVED** [review]

**Comment by rbradford** [comment]
> > Can we get the vm-virtio and vhost changes upstream first so we can depend on the rust-vmm versions instead of these forks?
> 
> @rbradford Either I did something wrong while creating the PR, or you're reviewing an interim commit, because with all changes applied, it should only depend on rust-vmm's crates. In fact, GitHub marks as _Outdated_ your comments on `Cargo.toml`. :-?

Yeh, no idea how that happened...but I can see it's all great now! Pleased to see the progress.

**Comment by alexandruag** [comment]
> We'd also need some volunteers to act as code owners. I volunteer myself. @sboeuf @rbradford @alexandruag @jiangliu would you be interested?

Thanks for considering, and I'll be happy to help whenever I'm able to, but I don't think I have sufficient context in this area yet to officially show up as a code owner :-s

File: Cargo.toml (hunk 3 above)
Comment (slp) [reply Cargo.toml]: Yeah, this is a good opportunity to get it right, so I fixed it. Thanks!

**Comment by rbradford** [comment]
@slp Can you add @sboeuf (he's away currently but he did write a lot of the code originally)

**Review by jiangliu — APPROVED** [review]

**Comment by slp** [comment]
> @slp Can you add @sboeuf (he's away currently but he did write a lot of the code originally)

Sure! Done.

**Review by sameo — APPROVED** [review]

**Comment by rbradford** [comment]
> > @slp Can you add @sboeuf (he's away currently but he did write a lot of the code originally)
> 
> Sure! Done.

Looks like it's still missing to me...but maybe this is my GitHub :ghost: again...

**Comment by slp** [comment]
> > > @slp Can you add @sboeuf (he's away currently but he did write a lot of the code originally)
> > 
> > 
> > Sure! Done.
> 
> Looks like it's still missing to me...but maybe this is my GitHub ghost again...

Nope, this time it's really missing (wrong force push). I'm creating a new PR.

**Pull request merged**